core/http-auth-aws-crt/src/main/java/software/amazon/awssdk/http/auth/aws/crt/internal/signer/AwsCrtV4aProperties.java

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.http.auth.aws.crt.internal.signer;
+
+import static software.amazon.awssdk.http.auth.aws.AwsV4HttpSigner.CHECKSUM_ALGORITHM;
+import static software.amazon.awssdk.http.auth.aws.AwsV4HttpSigner.CHECKSUM_HEADER_NAME;
+import static software.amazon.awssdk.http.auth.aws.crt.AwsCrtV4aHttpSigner.DOUBLE_URL_ENCODE;
+import static software.amazon.awssdk.http.auth.aws.crt.AwsCrtV4aHttpSigner.NORMALIZE_PATH;
+import static software.amazon.awssdk.http.auth.aws.crt.AwsCrtV4aHttpSigner.REGION_NAME;
+import static software.amazon.awssdk.http.auth.aws.crt.AwsCrtV4aHttpSigner.SERVICE_SIGNING_NAME;
+import static software.amazon.awssdk.http.auth.aws.crt.AwsCrtV4aHttpSigner.SIGNING_CLOCK;
+import static software.amazon.awssdk.http.auth.aws.util.CredentialUtils.sanitizeCredentials;
+import static software.amazon.awssdk.http.auth.aws.util.SignerUtils.validatedProperty;
```

Comment:
Seeing this use of cross-module dependency on a SdkProtectedApi for this, made me realize the validateProperty method has nothing to do with http-auth-aws specifically, and may be useful to other signers. Seems like something that could reside in a more generic place, probably in or along side an existing interface related to signers in http-auth-spi. Though probably to can be tackled with a TODO and done outside of this PR.

Reply:
Yeah, agreed. I can also just easily move it to http-auth-spi.